File: insertattail.js
//insert node at tail of linked list


/*
 * For your reference:
 *
 * SinglyLinkedListNode {
 *     int data;
 *     SinglyLinkedListNode next;
 * }
 *
 */
function insertNodeAtTail(head, data) {
    
    let current = head;
    const newNode = new SinglyLinkedListNode(data);
    if(head == null){
        head = newNode;
    }
    else {
        while(current.next !== null){
        current = current.next;
    }
    current.next = newNode;
    }
    return head;
}

File: insertattail.py
# insert node at tail of linked list


#*
#  * For your reference:
#  *
#  * SinglyLinkedListNode {
#  *     int data;
#  *     SinglyLinkedListNode next;
#  * }
#  *
#  *

def insertNodeAtTail(head, data):
    newnode = SinglyLinkedListNode(data)
    if head == None:
        head = newnode
        return head
    curr = head
    while curr.next:
        curr = curr.next

    curr.next = newnode
    return head
